lib/matplotlib/axes/_axes.py

```diff
@@ -5607,7 +5606,7 @@ def hist(self, x, bins=10, range=None, normed=False, weights=None,
         # basic input validation
         flat = np.ravel(x)
         if len(flat) == 0:
-            raise ValueError("x must have at least one data point")
+            return [], [], cbook.silent_list('No Patches')
```

Comment:
Do we want do draw _no_ patches or patches of zero size?  On consideration I think patches of zero size might be a better option here so client code which expects a non-zero length list does not have to check that it is true.  For example if the user passes in bin edges, they should get back a list of zeros telling them that there are no counts in any of the bins.